code/modules/mob/living/simple_animal/friendly/dog.dm

```diff
@@ -105,6 +105,17 @@
 	..(gibbed)
 	regenerate_icons()
 
+/mob/living/simple_animal/pet/dog/corgi/RangedAttack(atom/A, params)
+	if(inventory_back)
+		inventory_back.afterattack(A, src)
+
+/mob/living/simple_animal/pet/dog/corgi/UnarmedAttack(atom/A)
+	if(istype(inventory_back, /obj/item/extinguisher))
+		var/obj/item/extinguisher/E = inventory_back
+		E.AttemptRefill(A, src)
+		return
```

Comment:
I don't like how snowflakey this is. This will istype the inventory of a corgi everytime they attack something which is not very efficient. Is there anyway to do something similar to `RangedAttack` ?

Reply:
The only alternative i could think of is a `attack_animal` proc in the water tank void